platform/linux-generic/odp_ipsec.c

```diff
@@ -1892,6 +1893,10 @@ int _odp_ipsec_init_global(void)
 
 	memset(&default_out_opt, 0, sizeof(default_out_opt));
 
+	odp_global_data.ipsec_rand_kind = ODP_RANDOM_CRYPTO;
+	if (odp_global_data.ipsec_rand_kind > odp_random_max_kind())
+		odp_global_data.ipsec_rand_kind = odp_random_max_kind();
```

Comment:
If `odp_random_max_kind()` is less than `ODP_RANDOM_CRYPTO`, wouldn't it be better to simply say that IPsec is not available in this implementation? Providing poor crypto quality seems undesirable.

Reply:
This is used to generate IV. I would not use BASIC by default, but it is fine to be used if better random source is not available. Remember, before BEAST attack it was quite typical to send 'previous packet' as IV, which isn't even random.

Reply:
OK, we'll cover this caveat in the documentation.